packages/material-renderers/src/util/theme.ts

```diff
@@ -33,3 +40,29 @@ export interface JsonFormsTheme extends Theme {
     };
   };
 }
+
+export type WithInputProps = {
```

Comment:
Can we declare this as an interface?
To not break the API for components using this now, I think it could make sense to make `InputComponent` optional, too, and fall back to material's `Input` when it is not given. Do you think that makes sense?

Reply:
Looking again through the changes, I think the `InputComponent` prop can be eliminated completely by moving `useInputComponent()` down to each input control implementation. With that approach custom renderers leveraging the input controls should probably not break. What do you think?